src/solve/invariants.md

```diff
@@ -1,6 +1,8 @@
 # Invariants of the type system
 
+<!--
 FIXME: This file talks about invariants of the type system as a whole, not only the solver
+-->
```

Comment:
for example I think there is value in having this be user-visible, there's a potential "flaw" in the current document that is probably good for readers to be aware of and have explicitly called out